crypto/dh/dh_backend.c

```diff
@@ -183,6 +183,12 @@ DH *ossl_dh_dup(const DH *dh, int selection)
     return NULL;
 }
 
+void ossl_dh_clear_private_key(DH *dh)
+{
+    BN_clear_free(dh->priv_key);
+    dh->priv_key = NULL;
```

Comment:
Not sure if the length field also need clearing?

Reply:
Why would it? The length won't be used for public key operations but on the other hand I do not see a reason to clear it.